File: caesars_cipher.java
import java.util.Scanner;

public class Caesars_cipher {
    final String lower_case_alphabet = "abcdefghijklmnopqrstuvwxyz";
    final String upper_case_alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    final String digits = "1234567890";

    public static void main(String[] args) {
        //user input
        Scanner scanner = new Scanner(System.in);
        System.out.print("Input word or phrase to be encrypted: ");
        String user_input = scanner.nextLine();

        //get shift amount
        System.out.print("Enter shift amount: ");
        int shift = scanner.nextInt();

         //can be updated to account for input numbers 
        System.out.println("User input -->      " + user_input);
        Caesars_cipher cipher = new Caesars_cipher();

        System.out.println("Encrypted code -->     " + cipher.encrypt(user_input, shift));
    }
    public String encrypt(String input, int s) {
        int current, rank;
        StringBuffer encoded = new StringBuffer();
        char c;

        //iterate through input to encrypt
        for (int i=0; i<input.length(); i++) {
            c = input.charAt(i);
            // if uppercase, use uppercase alphabet
            if (Character.isUpperCase(c)) {
                // c = nth char of text input
                // calculate r = rank of c in alphabet
                current = upper_case_alphabet.indexOf(input.charAt(i));
                // shift and get position within alphabet
                rank = (current + s) % 26;
                //letter with rank R2 in alphabet
                c = upper_case_alphabet.charAt(rank);
                encoded.append(c);
            //else if lowercase, use lowercase alphabet
            } else if (Character.isLowerCase(c)) {
                // c = nth char of text input
                // calculate r = rank of c in alphabet
                current = lower_case_alphabet.indexOf(input.charAt(i));
                // shift and get position within alphabet
                rank = (current + s) % 26;
                //letter with rank R2 in alphabet
                c = lower_case_alphabet.charAt(rank);
                encoded.append(c);
            //else if a digit, iterate over 10 numbers 0 - 9
            //} else if (Character.isDigit(input.charAt(c))) {
                // c = nth char of text input
                // calculate r = rank of c in alphabet
                //current = digits.indexOf(input.charAt(i));
                // shift and get position within alphabet
                //rank = (current + s) % 10;
                //letter with rank R2 in alphabet
                //c = digits.charAt(rank);
                //encoded.append(c);
            //else , if not upper or lowercase, leave as is 
            } else {
                encoded.append(c);
            }

        }
        return encoded.toString();
    }
}

File: caesars_cipher.py
#!/usr/bin/python3
# encrypt and decrypt arbitrary string (and ints) using Ceasar shift cipher

text = input("Enter message to encrypt: ")
shift = input("Enter shift: ")
shift = int(shift)

def encrypt(string, s):
    encryption = ""
    #traverse each char in text
    for c in text:
        # if uppercase, use uppercase ord()
        if c.isupper():
            # ord() returns an int representing the Unicode char
            # to find the position in 0-25
            c_unicode = ord(c)
            c_index = ord(c) - ord('A')
            #shift
            new_index = (c_index + shift) % 26
            #convert to new char
            new_unicode = new_index + ord('A')
            new_char = chr(new_unicode)
            encryption += new_char

        # if lowercase, use lowercase ord()
        elif c.islower():
            c_unicode = ord(c)
            c_index = ord(c) - ord('a')
            #shift
            new_index = (c_index + shift) % 26
            #convert to new char
            new_unicode = new_index + ord('a')
            new_char = chr(new_unicode)
            encryption += new_char

        # if a number, iterate through 9 numbers
        elif c.isdigit():
            c_unicode = ord(c)
            c_index = ord(c) - ord('0')
            #shift
            new_index = (c_index + shift) % 10
            #convert to new char
            new_unicode = new_index + ord('0')
            new_char = chr(new_unicode) 
            encryption += new_char

        else:
            #if not uppercase or lowercase, leave as is
            encryption += c
            #could shift for numbers in text
    return encryption

print("\nEncryption------->")
print("Plain text: ", text)
print("Shift\t  : ", str(shift))
print("Encrypted : ", encrypt(text,shift))

encrypted_text = encrypt(text,shift)

def decrypt(string):
    decryption = ""
    for c in encrypted_text:
    # if uppercase, use uppercase ord()
        if c.isupper():
            # ord() returns an int representing the Unicode char
            # to find the position in 0-25
            c_unicode = ord(c)
            c_index = ord(c) - ord('A')
            #shift
            new_index = (c_index - shift) % 26
            #convert to new char
            new_unicode = new_index + ord('A')
            new_char = chr(new_unicode)
            decryption += new_char

        # if lowercase, use lowercase ord()
        elif c.islower():
            c_unicode = ord(c)
            c_index = ord(c) - ord('a')
            #shift
            new_index = (c_index - shift) % 26
            #convert to new char
            new_unicode = new_index + ord('a')
            new_char = chr(new_unicode) 
            decryption += new_char
        # if a number, iterate through 9 numbers
        elif c.isdigit():
            c_unicode = ord(c)
            c_index = ord(c) - ord('0')
            #shift
            new_index = (c_index - shift) % 10
            #convert to new char
            new_unicode = new_index + ord('0')
            new_char = chr(new_unicode) 
            decryption += new_char
        else:
            #if not uppercase or lowercase, leave as is
            decryption += c
            #could shift for numbers in text
    return decryption

#to do: ask user if they want to decrypt
print("\nDecryption------->")
print("Encrypted : ", encrypted_text)
print("Shift\t  : ", str(shift))
print("Decrypted : ", decrypt(encrypted_text))

#output every possible shift decryption
def brute_force(string):
    #letter shift index
    x = 0
    #number shift index
    lx = 0
    print("Brute force decryption------->")
    while x < 26:
        attempt = ""
        x = x+1
        lx = x+1
        for c in encrypted_text:
            if c.isupper():
                c_unicode = ord(c)
                c_index = ord(c) - ord('A')
                #shift 1
                new_index = (c_index + x) % 26
                #convert to new char
                new_unicode = new_index + ord('A')
                new_char = chr(new_unicode)
                attempt += new_char
            # if lowercase, use lowercase ord()
            elif c.islower():
                c_unicode = ord(c)
                c_index = ord(c) - ord('a')
                #shift
                new_index = (c_index + x) % 26
                #convert to new char
                new_unicode = new_index + ord('a')
                new_char = chr(new_unicode) 
                attempt += new_char
            # if a number, iterate through 9 numbers
            elif c.isdigit():
                c_unicode = ord(c)
                c_index = ord(c) - ord('0')
                #shift
                new_index = (c_index + lx) % 10
                #convert to new char
                new_unicode = new_index + ord('0')
                new_char = chr(new_unicode) 
                attempt += new_char
            else:
                attempt += c

        print(x,":\t",attempt)
    return attempt

user_input = input("\nEnter 'Y/y' to see all possible decryptions or any key to exit: ")
if (user_input == 'y') or (user_input == 'Y'):
    brute_force(encrypted_text)






